File: android/src/main/java/com/jjlf/rngradient/SweepGradientViewManager.java
package com.jjlf.rngradient;

import android.graphics.Color;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.ViewGroupManager;
import com.facebook.react.uimanager.ViewProps;
import com.facebook.react.uimanager.annotations.ReactProp;

public class SweepGradientViewManager  extends ViewGroupManager<SweepGradientView> {


    @Override
    public String getName() { return "ConicGradient"; }


    @Override
    protected SweepGradientView createViewInstance(@NonNull ThemedReactContext reactContext) {
        return new SweepGradientView(reactContext);
    }

    @Override
    @ReactProp(
            name = ViewProps.BACKGROUND_COLOR,
            defaultInt = Color.TRANSPARENT,
            customType = "Color")
    public void setBackgroundColor(@NonNull SweepGradientView view, int backgroundColor) { }
    @ReactProp(name = "startPoint")
    public void setStartPoint(SweepGradientView view , ReadableMap m){}
    @ReactProp(name = "endPoint")
    public void setEndPoint(SweepGradientView view , ReadableMap m){}

    @ReactProp(name = "centerX",defaultFloat = 0.5f)
    public void setCenterX(SweepGradientView view , float v){
        view.getDrawable().setCenterX(v);
        view.getDrawable().invalidateSelf();
    }
    @ReactProp(name = "centerY",defaultFloat = 0.5f)
    public void setCenterY(SweepGradientView view , float v){
        view.getDrawable().setCenterY(v);
        view.getDrawable().invalidateSelf();
    }
    @ReactProp(name = "conicRotation",defaultFloat = 0f)
    public void setConicRotation(SweepGradientView view,float v){
        view.getDrawable().setSweepRotation(v);
        view.getDrawable().invalidateSelf();
    }

    @ReactProp(name = "colors")
    public void setColors(SweepGradientView view, ReadableArray colors)  {
        int[] c = colors != null ? ModUtil.toIntArray(colors) : new int[] { Color.WHITE, Color.BLACK};
        view.getDrawable().setColors(c);
        view.getDrawable().invalidateSelf();
    }
    @ReactProp(name = "positions")
    public void setPositions(SweepGradientView view, ReadableArray positions)  {
        float[] p = positions != null ? ModUtil.toFloatArray(positions) : null;
        view.getDrawable().setPositions(p);
        view.getDrawable().invalidateSelf();
    }
}

File: android/src/main/java/com/jjlf/rngradient/SweepGradientViewManager.kt
package com.jjlf.rngradient

import android.content.Context
import android.graphics.Color
import com.facebook.react.bridge.ReadableArray
import com.facebook.react.bridge.ReadableMap
import com.facebook.react.uimanager.annotations.ReactProp
import com.facebook.react.uimanager.*
import com.jjlf.rngradient.GradientHelper
import com.jjlf.rngradient.LinearGradientView
import com.jjlf.rngradient.SweepGradientView


class SweepGradientViewManager : ViewGroupManager<SweepGradientView>() {

    override fun getName(): String {
        return "ConicGradient"
    }

    override fun createViewInstance(reactContext: ThemedReactContext): SweepGradientView {
        return SweepGradientView(reactContext as Context)
    }

    @Override
    @ReactProp(name = ViewProps.BACKGROUND_COLOR, defaultInt = Color.TRANSPARENT, customType = "Color")
    override fun setBackgroundColor(view: SweepGradientView, c:Int){}

    @ReactProp(name = "startPoint")
    fun setStartPoint(view: LinearGradientView, m:ReadableMap?){}
    @ReactProp(name = "endPoint",)
    fun setEndPoint(view: LinearGradientView, m:ReadableMap?){}

    @ReactProp(name = "centerX",defaultFloat = 0.5f)
     fun setCenterX(view: SweepGradientView, v:Float){
         val mDrawable = view.getDrawable()
        mDrawable.setCenterX(v)
        mDrawable.invalidateSelf()

    }
    @ReactProp(name = "centerY",defaultFloat = 0.5f)
    fun setCenterY(view: SweepGradientView, v:Float){
        val mDrawable = view.getDrawable()
        mDrawable.setCenterY(v)
        mDrawable.invalidateSelf()
    }
    @ReactProp(name = "conicRotation",defaultFloat = 0f)
    fun setConicRotation(view: SweepGradientView, v:Float){
        val mDrawable = view.getDrawable()
        mDrawable.setSweepRotation(v)
        mDrawable.invalidateSelf()
    }

    @ReactProp(name = "colors")
    fun setColors(view: SweepGradientView, colors: ReadableArray?)  {
        val mDrawable = view.getDrawable()
        val c = if(colors != null) GradientHelper.toIntArray(colors) else intArrayOf(
            Color.WHITE,
            Color.BLACK)
        mDrawable.setColors(c)
        mDrawable.invalidateSelf()
    }
    @ReactProp(name = "positions")
    fun setPositions(view: SweepGradientView, positions: ReadableArray?)  {
        val mDrawable = view.getDrawable()
        val p = if(positions != null) GradientHelper.toFloatArray(positions) else null
        mDrawable.setPositions(p)
        mDrawable.invalidateSelf()
    }

}